deploy/lib/createContainers.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+const constants = require('./constants');
+
+module.exports = {
+  createContainers() {
+    return BbPromise.bind(this)
+      .then(this.getContainers)
+      .then(this.createOrUpdateContainers);
+  },
+
+  getContainers() {
+    const containersUrl = `namespaces/${this.namespace.id}/containers`;
+    return this.provider.apiManager.get(containersUrl)
+      .then(response => response.data.containers)
+      .catch((err) => {
+        throw new Error(err.response.data.message)
+      })
+  },
+
+  createOrUpdateContainers(foundContainers) {
+
+    const containers = this.provider.serverless.service.custom;
+    const containerNames = Object.keys(containers);
+    const promises = containerNames.map((containerName) => {
+      const cont = Object.assign(containers[containerName], { name: containerName });
+      const foundCont = foundContainers.find(c => c.name === cont.name);
+      return foundCont ? this.updateContainer(cont, foundCont) : this.createSingleContainer(cont);
+    });
+
+    return Promise.all(promises)
+      .then((updatedContainers) => {
+        this.containers = updatedContainers;
+      });
+  },
+
+  createSingleContainer(cont) {
+    const params = {
+      name: cont.name,
+      environment_variables: cont.env,
+      namespace_id: this.namespace.id,
+      memory_limit: cont.memoryLimit,
+      min_scale: cont.minScale,
+      max_scale: cont.maxScale,
+      timeout: cont.timeout
+    };
+
+    this.serverless.cli.log(`Creating container ${cont.name}...`);
+
+    return this.provider.apiManager.post('containers', params)
+      .then(response => Object.assign(response.data, { directory: cont.directory }))
+      .catch((err) => {
+        throw new Error(err.response.data.message)
+      })
+  },
+
+  updateContainer(cont, foundCont) {
+
+    const params = {};
```

Comment:
I believe it would be cleaner to just do
```javascript
const params = {
  redeploy: false,
  environment_variables: container.env,
  memory_limit: container.memoryLimit,
  // ...
};
```